### PR TITLE
Refreshing devices list after adding a disk or cdrom controller

### PR DESCRIPTION
**Pull request by dkalleg**
Without refreshing the devices list, we can't find the controller we just created.

### CODE REVIEW

**Comment by dkalleg** [comment]
@chrislovecnm @stack72 @phinze 
Adding folks for visibility, please review at your convenience.

File: builtin/providers/vsphere/resource_vsphere_virtual_machine.go [hunk 1]
```diff
@@ -1153,9 +1153,14 @@ func addHardDisk(vm *object.VirtualMachine, size, iops int64, diskType string, d
...
 		vm.AddDevice(context.TODO(), c)
+		// Update our devices list
+		devices, err := vm.Device(context.TODO())
+		if err != nil {
+			return err
+		}
 		controller, err = devices.FindDiskController(controller_type)
 		if err != nil {
-			return fmt.Errorf("[ERROR] Could not find the controller we just created")
+			return fmt.Errorf("[ERROR] Could not find the new %v controller: %v", controller_type, err)
```
Comment (chrislovecnm) [inline builtin/providers/vsphere/resource_vsphere_virtual_machine.go]: please log `err`, or check that it is logged in stack above.

**Comment by chrislovecnm** [comment]
@dkalleg is this an edge case or should have we caught this in testing?

**Comment by thetuxkeeper** [comment]
@dkalleg : Interesting. It's the same way I do it in #7154 for the network interfaces. Little sad that we have to spam api calls (create device function with created device as return value would be great)
@chrislovecnm : Since it's all before the `Read()` I think it's "just" a bug. the tfstate will include the created controller, so the tests won't see it.
Additionally I think (not tested) it will only fail if we add a controller type that wasn't present before (will find the old/wrong one). So we need a test that adds a non-default controller (I think a VM has a controller automatically, should be scsi) or make sure that we have no controllers at all and add one.

**Comment by dkalleg** [comment]
@stack72 @phinze @thetuxkeeper @chrislovecnm 
Any actionable feedback?

**Comment by chrislovecnm** [comment]
LGTM!!

**Comment by dkalleg** [comment]
Thanks @stack72

**Pull request merged**